src/db/thumbnails.c

```diff
@@ -230,27 +230,39 @@ dt_thumbnails_cache_one(
   char deffilename[PATH_MAX+100];
   char bc1filename[PATH_MAX+100];
   uint64_t hash = hash64(filename);
-  snprintf(bc1filename, sizeof(bc1filename), "%s/%lx.bc1", tn->cachedir, hash);
+  snprintf(bc1filename, sizeof(bc1filename), "%s/%llx.bc1", tn->cachedir, hash);
   snprintf(cfgfilename, sizeof(cfgfilename), "%s", filename);
   snprintf(deffilename, sizeof(deffilename), "default.%"PRItkn, dt_token_str(input_module));
   struct stat statbuf = {0};
   time_t tcfg = 0, tbc1 = 0;
 
   if(!stat(cfgfilename, &statbuf))
+#ifndef __APPLE__
```

Comment:
okay we probably want to factor this out in something like `time_t t = dt_stat(filename)` defined in `core.h` or similar, so the platform specific code is only in one place.

Reply:
Yes I will do something like that to put together all plat-spec code. Thank you for the tip 😉